R/slice_derivation.R

```diff
@@ -156,10 +156,10 @@ slice_derivation <- function(dataset,
 #'
 #' @return An object of class `slice`.
 derivation_slice <- function(filter,
-                             args) {
+                             args = NULL) {
   out <- list(
     filter = assert_filter_cond(enquo(filter)),
-    args = assert_s3_class(args, "params")
+    args = assert_s3_class(args, "params", optional = TRUE)
```

Comment:
In the documentation- should we mention that an object of s3 class is needed for `dervivation_slice()`

Reply:
Are the tests for `assert_s3_class` in `admiraldev` enough for the optional = TRUE.  I didn't really look into that part, but seems like we might need some additional testing

Reply:
The documentation says that a `params` object is expected. I don't think that we need to mention that `params` objects are S3 class objects because this information does not help the users.

Test 39 is testing "no error if `arg` is NULL and optional is TRUE". Do you think we need a test for "error if `arg` is NULL and optional is FALSE"?

Reply:
Understood on documentation.  I think an additional test is nice to have.

Reply:
Added tests in https://github.com/pharmaverse/admiraldev/pull/205.